src/api/java/io/github/cvc5/Proof.java

```diff
@@ -20,39 +20,28 @@
 /**
  * A cvc5 Proof.
  */
-public class Proof implements IPointer
+public class Proof extends AbstractPointer implements Comparable<Proof>
 {
   /**
    * Null proof
    */
   public Proof()
   {
-    this(getNullProof());
+    super(getNullProof());
   }
 
   private static native long getNullProof();
 
   Proof(long pointer)
   {
-    this.pointer = pointer;
```

Comment:
Keep this code

Reply:
This should be covered by the super call.